lib/orbit-common/cache/query-operators.js

```diff
@@ -0,0 +1,88 @@
+import { merge } from 'orbit/lib/objects';
+import { queryExpression as oqe } from 'orbit/query/expression';
+import { splitPath } from 'orbit/lib/paths';
+
+export default {
+  and(context, ...expressions) {
+    for (let expression of expressions) {
+      if (!this.evaluate(expression, context)) {
+        return false;
+      }
+    }
+    return true;
+  },
+
+  or(context, ...expressions) {
+    for (let expression of expressions) {
+      if (!!this.evaluate(expression, context)) { return true; }
+    }
+    return false;
+  },
+
+  equal(context, ...expressions) {
+    let value;
+    let valueSet = false;
+
+    for (let expression of expressions) {
+      if (!valueSet) {
+        value = this.evaluate(expression, context);
+        valueSet = true;
+      } else if (value !== this.evaluate(expression, context)) {
+        return false;
+      }
+    }
+
+    return true;
+  },
+
+  get(context, path) {
+    path = splitPath(path);
+    if (context.basePath) {
+      path = context.basePath.concat(path);
+    }
+    return this.target.get(path);
```

Comment:
My thinking was that every query evaluator would have a target source. It could also be on `context`, but then you'd have to know to add it to every context.